src/app_service/service/community/service.nim

```diff
@@ -1048,7 +1048,9 @@ QtObject:
       self.events.emit(SIGNAL_COMMUNITY_SPECTATED, CommunityArgs(community: updatedCommunity, fromUserAction: true, isPendingOwnershipRequest: (ownerTokenNotification != nil)))
 
       for k, chat in updatedCommunity.chats:
-        let fullChatId = communityId & chat.id
+        var fullChatId = chat.id
+        if not chat.id.startsWith(communityId):
+          fullChatId = communityId & chat.id
```

Comment:
I think this code never worked before. The chat ID returned was the long one with the community ID prefixed, so we were double prefixing. Because of that `asyncLoadInitialMessagesForChat` (called below) would always fail